src/util/merkleblock.rs

```diff
@@ -440,18 +440,31 @@ impl MerkleBlock {
     /// # }
     /// ```
     pub fn from_block(block: &Block, match_txids: &HashSet<Txid>) -> Self {
-        let header = block.header;
+        let block_txids: Vec<_> = block.txdata.iter().map(Transaction::txid).collect();
+        Self::from_header_txids(&block.header, &block_txids, match_txids)
+    }
+
+    /// Create a MerkleBlock from the block's header and txids, that should contain proofs for match_txids.
+    ///
+    /// The `header` is the block header, `block_txids` is the full list of txids included in the block and
+    /// `match_txids` is a set containing the transaction ids that should be included in the partial merkle tree.
+    /// ```
 
-        let mut matches: Vec<bool> = Vec::with_capacity(block.txdata.len());
-        let mut hashes: Vec<Txid> = Vec::with_capacity(block.txdata.len());
+    pub fn from_header_txids(
+        header: &BlockHeader,
+        block_txids: &[Txid],
+        match_txids: &HashSet<Txid>,
```

Comment:
Hmm, it would be a breaking change, but this could be a closure `Fn(&Txid) -> bool`. I know that there is at least one popular alternative `HashSet` implementation that a user could be using.

Are we planning to make a patch release with this? If not, it might be worth changing.